src/couch/include/couch_db.hrl

```diff
@@ -219,3 +219,29 @@
 -type sec_props() :: [tuple()].
 -type sec_obj() :: {sec_props()}.
 
+%% Erlang/OTP 21 deprecates and 23 removes get_stacktrace(), so 
+%% we have to monkey around until we can drop support < 21.
+%% h/t https://github.com/erlang/otp/pull/1783#issuecomment-386190970
+
+%% use like so:
+% try function1(Arg1)
+% catch
+%     ?STACKTRACE(exit, badarg, ErrorStackTrace)
+%         % do stuff with ErrorStackTrace
+%         % ...
+% end,
+
+% for features specific to Erlang/OTP version 20.x (and later versions)
+-ifdef(OTP_MAJOR_VSN_20).
+-else.
+-define(ERLANG_OTP_VERSION_21_FEATURES, true).
+-endif.
```

Comment:
There might be an easier way to detect 20 vs 21+ we already do it in fabric2 https://github.com/apache/couchdb/blob/00811b4a9c4df56defb7456431d44abb7c0baa79/src/fabric/src/fabric2_util.erl#L386-L387